WordPress/Classes/ViewRelated/Blog/BlogDetailsViewController.m

```diff
@@ -1474,20 +1474,27 @@ - (void)handleDataModelChange:(NSNotification *)note
     }
 
     NSUInteger generalSectionCountBefore = self.tableSections[0].rows.count;
+    BlogDetailsSection *firstSectionBefore = [self.tableSections objectAtIndex:0];
 
     NSSet *updatedObjects = note.userInfo[NSUpdatedObjectsKey];
     if ([updatedObjects containsObject:self.blog] || [updatedObjects containsObject:self.blog.settings]) {
         self.navigationItem.title = self.blog.settings.name;
         [self configureTableViewData];
         NSUInteger generalSectionCountAfter = self.tableSections[0].rows.count;
+        BlogDetailsSection *firstSectionAfter = [self.tableSections objectAtIndex:0];
+
         // quick start was just enabled
-        if (generalSectionCountBefore != generalSectionCountAfter && [self shouldShowQuickStartChecklist]) {
-            if ([Feature enabled:FeatureFlagQuickStartV2]) {
+
+        if ([Feature enabled:FeatureFlagQuickStartV2]) {
+            if (!firstSectionBefore.showQuickStartMenu && firstSectionAfter.showQuickStartMenu) {
```

Comment:
I was thinking if we can use one if statement only here like `[Feature enabled:FeatureFlagQuickStartV2] && !firstSectionBefore.showQuickStartMenu && firstSectionAfter.showQuickStartMenu`, what do you think?

Reply:
This would change the logic in the way that if `!firstSectionBefore.showQuickStartMenu` or `firstSectionAfter.showQuickStartMenu` are false, then the `else if` will evaluate as normal. I only want `if (generalSectionCountBefore != generalSectionCountAfter && [self shouldShowQuickStartChecklist])` evaluated if the feature flag is off. I think merging the if statements would confuse that separation.